onnxruntime/core/mlas/lib/pooling.cpp

```diff
@@ -1231,9 +1251,8 @@ Return Value:
     //
 
     //TODO: use a safeint here and make sure the result value can fit into int32_t
-    size_t TotalChannelCount = size_t(InputShape[0]) * size_t(InputShape[1]);
+    int32_t TotalChannelCount = int32_t(InputShape[0] * InputShape[1]);
```

Comment:
The original line was the correct way to do this consistent with the rest of MLAS. The dense tensor exists in memory so its size can be cast to size_t. And the multiply of two size_t's is okay here again because the tensor exists. The TODO should be removed, somebody not me put that there.